lib/autoproj/daemon/git_api/services/github.rb

```diff
@@ -53,7 +67,35 @@ def initialize(**options)
                     def pull_requests(git_url, **options)
                         exception_adapter do
                             @client.pull_requests(git_url.path, **options).map do |pr|
-                                PullRequest.from_ruby_hash(git_url, pr.to_hash)
+                                mergeable =
+                                    if pr["merged_at"]
+                                        false
+                                    else
+                                        pull_request_mergeable?(git_url, pr["number"])
+                                    end
+
+                                pr = pr.to_hash.merge({ "mergeable" => mergeable })
+                                PullRequest.from_ruby_hash(git_url, pr)
+                            end
+                        end
+                    end
+
+                    # Check if a pull request is mergeable
+                    #
+                    # @param [GitAPI::URL] repo_url the repository URL
+                    # @param [Integer] number the pull request number
+                    def pull_request_mergeable?(git_url, number, poll: 0.1)
+                        loop do
```

Comment:
I think you should add a timeout so that the daemon doesn't hang forever if `mergeable` doesn't become available for whatever reason (ie. github bug). I've personally seen this happen on gitlab. Also, it would be good to add some kind of output so one can keep track of what's happening in the logs (i.e how long it waited for the `mergeable` status if one wasn't available after a reasonable amount of time)

Reply:
Ack on the timeout. I should return false then I guess ?

There's the "waiting for ..." message. That's how I found out I had to remove the cache.

Reply:
> I should return false then I guess ?

Sounds reasonable.

> There's the "waiting for ..." message.

I missed that somehow.